network-api/networkapi/wagtailpages/pagemodels/libraries/rcc/detail_page.py

```diff
@@ -115,64 +107,148 @@ class RCCDetailPage(BasePage):
+        context["rcc_authors"] = self.get_rcc_authors()
+        context["rcc_author_names"] = self.get_rcc_author_names()
+        context["content_type_names"] = self.get_related_content_types_names()
+        return context
 
-    # def get_rcc_authors(self):
-    #     rcc_author_profiles = localize_queryset(
-    #         Profile.objects.prefetch_related("authored_rcc_entries").filter(authored_rcc_entries__rcc_detail_page=self)
-    #     )
-    #     return rcc_author_profiles
+    def get_rcc_authors(self):
+        rcc_author_profiles = wagtailpages_utils.localize_queryset(
+            profiles.Profile.objects.prefetch_related("authored_rcc_articles").filter(
+                authored_rcc_articles__rcc_detail_page=self
+            )
+        )
+        return rcc_author_profiles
 
-    # def get_rcc_author_names(self):
-    #     return [ra.author_profile.name for ra in self.rcc_authors.all()]
+    def get_rcc_author_names(self):
+        return [ra.author_profile.name for ra in self.rcc_authors.all()]
 
-    # def get_related_content_types_names(self):
-    #     return [ct.content_type.name for ct in self.related_content_types.all()]
+    def get_related_content_types_names(self):
+        return [ct.content_type.name for ct in self.related_content_types.all()]
 
     def get_banner(self):
         return self.get_parent().specific.get_banner()
+
+
+class RCCDetailLink(wagtail_models.TranslatableMixin, wagtail_models.Orderable):
+    rcc_detail_page = cluster_fields.ParentalKey(
+        "RCCDetailPage",
+        null=False,
+        blank=False,
+        on_delete=models.CASCADE,
+        related_name="rcc_links",
+    )
+
+    label = models.CharField(null=False, blank=False, max_length=50)
+
+    url = models.URLField(null=False, blank=True)
+    page = models.ForeignKey(
+        "wagtailcore.Page",
+        null=True,
+        blank=True,
+        on_delete=models.CASCADE,
+    )
+    document = models.ForeignKey(
+        wagtail_docs.get_document_model_string(),
+        null=True,
+        blank=True,
+        on_delete=models.CASCADE,
+    )
+
+    panels = [
+        edit_handlers.HelpPanel(
+            content=(
+                "Please provide a link to the original resource. "
+                "You can link to an internal page, an external URL or upload a document. "
+                'If you wish to provide multiple, please create two separate "article links"'
+            )
+        ),
+        edit_handlers.FieldPanel("label"),
+        edit_handlers.FieldPanel("url"),
+        edit_handlers.FieldPanel("page"),
+        edit_handlers.FieldPanel("document"),
+    ]
+
+    class Meta(wagtail_models.TranslatableMixin.Meta, wagtail_models.Orderable.Meta):
+        ordering = ["sort_order"]
+
+    def __str__(self) -> str:
+        return self.label
+
+    def clean(self) -> None:
+        super().clean()
+
+        url_set = bool(self.url)
+        page_set = bool(self.page)
+        document_set = bool(self.document)
```

Comment:
I thought I mentioned this somewhere already, but what do you think of renaming this to `is_url_set`, etc., a more question style naming. `_set` sounds like multiple. I.e. in Django it's also the default for backward relationships. 

I  also like this one: https://github.com/kettanaito/naming-cheatsheet